src/watchmaker/__init__.py

```diff
@@ -452,6 +452,12 @@ def install(self):
             self.log.critical(msg)
             raise
 
+        watchmaker.utils.set_file_perms(self.system_params['prepdir'])
```

Comment:
Did you happen to try setting the mode on `self.system_params['prepdir']` just after

```
            os.makedirs(self.system_params['workingdir'])
```
on line 433?

I'm curious if the files would all just inherit the correct perms then, so we wouldn't need to loop over all the files/dirs to set them individually...

Reply:
It won't work. On nix, there's no inheritance but I'll set the umask before anything is created (ln 410). On doze, oschmod specifically disables inheritance to improve security/greatly simplify permissions.